kymatio/scattering1d/core/timefrequency_scattering1d.py

```diff
@@ -0,0 +1,1314 @@
+import math
+from ..backend.agnostic_backend import unpad_dyadic
+
+
+def timefrequency_scattering1d(
+        x, pad, unpad, backend, J, log2_T, psi1, psi2, phi, scf,
+        pad_fn, pad_left=0, pad_right=0, ind_start=None, ind_end=None,
+        oversampling=0, oversampling_fr=0, aligned=True, average=True,
+        average_global=None, average_global_phi=None, out_type='array',
+        out_3D=False, out_exclude=None, pad_mode='zero'):
+    """
+    Main function implementing the Joint Time-Frequency Scattering transform.
+
+    Below is implementation documentation for developers.
+
+    Frequential scattering
+    ======================
+
+    Variables
+    ---------
+    Explanation of variable naming and roles. For `scf`'s attributes see
+    full Attributes docs in `TimeFrequencyScattering1D`.
+
+    subsample_equiv_due_to_pad
+        Equivalent amount of subsampling due to padding, relative to
+        `J_pad_frs_max_init`.
+        Distinguishes differences in input lengths (to `_frequency_scattering()`
+        and `_joint_lowpass()`) due to padding and conv subsampling.
+        This is needed to tell whether input is *trimmed* (pad difference)
+        or *subsampled*.
+
+    "conv subsampling"
+        Refers to `subsample_fourier()` after convolution (as opposed to
+        equivalent subsampling due to padding). i.e. "actual" subsampling.
+        Refers to `total_conv_stride_over_U1`.
+
+    total_conv_stride_over_U1
+        See above. Alt name: `total_conv_stride_fr`; `over_U1` seeks
+        to emphasize that it's the absolute stride over first order coefficients
+        that matters rather than equivalent/relative quantities w.r.t. padded etc
```

Comment:
> the absolute stride over first order coefficients that matters rather than equivalent/relative quantities w.r.t. padded

you're making it sound like stride depends upon padding? is this another `aligned=False` trick?